mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java

```diff
@@ -112,6 +119,63 @@ public CassandraMessageIdMapper(MailboxMapper mailboxMapper, CassandraMailboxDAO
             .sorted(Comparator.comparing(MailboxMessage::getUid));
     }
 
+    private CompletableFuture<Stream<Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>>>>
+            retrieveMessages(FetchType fetchType, ImmutableList<ComposedMessageIdWithMetaData> composedMessageIds) {
+        return messageDAOV2.retrieveMessages(composedMessageIds, fetchType, UNLIMITED)
+                .thenCompose(messageResults -> FluentFutureStream.of(messageResults
+                        .map(this::handleVersioning))
```

Comment:
you should make that clear that you load it from v1 as a fallback, `handleVersioning` is not explicit enough IMO